crates/bevy_ecs/src/system/commands/command_queue.rs

```diff
@@ -53,11 +56,16 @@ impl CommandQueue {
         }
 
         let meta = CommandMeta {
-            apply_command_and_get_size: |command, world| {
+            consume_command_and_get_size: |command, world| {
                 // SAFETY: According to the invariants of `CommandMeta.apply_command_and_get_size`,
                 // `command` must point to a value of type `C`.
                 let command: C = unsafe { command.read_unaligned() };
-                command.apply(world);
+                match world {
+                    // Apply command to the provided world...
+                    Some(world) => command.apply(world),
+                    // ...or discard it.
+                    None => drop(command),
```

Comment:
An explicit `drop` call is not required, since it will be implicitly called once the scope of this closure ends. I've added it here purely for clarity of purpose.